### PR TITLE
Issue #304: Use shared PNG icons

### LINKED ISSUE #304: Update filter icon of Checkstyle to the latest Eclipse filter icon
Would you accept a PR which updates the Checkstyle filter icon to the latest from the Eclipse platform?

image

### PR DESCRIPTION
**Pull request by Bananeweizen**
Fixes #304 and #305 by using the platform icons instead of local copies.

### CODE REVIEW

**Comment by rnveach** [comment]
https://app.travis-ci.com/github/checkstyle/eclipse-cs/jobs/588029921
````
Caused by: java.lang.NoSuchMethodError: 'com.puppycrawl.tools.checkstyle.api.LocalizedMessage com.puppycrawl.tools.checkstyle.TreeWalkerAuditEvent.getLocalizedMessage()'
	at com.puppycrawl.tools.checkstyle.filters.JavaPatchFilterElement.isLineMatching(JavaPatchFilterElement.java:217)
	at com.puppycrawl.tools.checkstyle.filters.JavaPatchFilterElement.accept(JavaPatchFilterElement.java:174)
	at com.puppycrawl.tools.checkstyle.filters.SuppressionJavaPatchFilter.accept(SuppressionJavaPatchFilter.java:199)
	at com.puppycrawl.tools.checkstyle.TreeWalker.getFilteredViolations(TreeWalker.java:187)
	at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:165)
	at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:87)
	at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:331)
	at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:292)
	... 27 more

Exception in thread "Timer-2" java.lang.NoClassDefFoundError: org/eclipse/osgi/internal/messages/Msg
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.setProperties(ServiceRegistrationImpl.java:174)
	at org.apache.felix.scr.impl.ComponentRegistry$4.run(ComponentRegistry.java:751)
	at java.base/java.util.TimerThread.mainLoop(Timer.java:566)
	at java.base/java.util.TimerThread.run(Timer.java:516)
Caused by: java.lang.ClassNotFoundException: org.eclipse.osgi.internal.messages.Msg
	at org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy.loadClass(SelfFirstStrategy.java:50)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.unsynchronizedLoadClass(ClassRealm.java:271)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:247)
	at org.codehaus.plexus.classworlds.realm.ClassRealm.loadClass(ClassRealm.java:239)
````

**Comment by rnveach** [comment]
@romani ping
> Caused by: java.lang.NoSuchMethodError: 'com.puppycrawl.tools.checkstyle.api.LocalizedMessage com.puppycrawl.tools.checkstyle.TreeWalkerAuditEvent.getLocalizedMessage()'
> at com.puppycrawl.tools.checkstyle.filters.JavaPatchFilterElement.isLineMatching(JavaPatchFilterElement.java:217)

This looks like an incompatible version I was talking to you about. I don't know why master isn't failing here, but since patch relies on 8.35 and eclipse-cs uses 10, these 2 don't seem compatible.

File: net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/CheckstyleUIPluginImages.java [hunk 1]
```diff
@@ -101,8 +104,7 @@ public abstract class CheckstyleUIPluginImages {
             "icons/checkstyle_warning.gif"); //$NON-NLS-1$
     MARKER_INFO = AbstractUIPlugin.imageDescriptorFromPlugin(CheckstyleUIPlugin.PLUGIN_ID,
             "icons/checkstyle_info.gif"); //$NON-NLS-1$
-    HELP_ICON = AbstractUIPlugin.imageDescriptorFromPlugin(CheckstyleUIPlugin.PLUGIN_ID,
-            "icons/help.gif"); //$NON-NLS-1$
+    HELP_ICON = PlatformUI.getWorkbench().getSharedImages().getImageDescriptor(ISharedImages.IMG_LCL_LINKTO_HELP);
```
Comment (rnveach) [inline net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/CheckstyleUIPluginImages.java:107]: Greater than 100 lines. Also see the PR discussion about putting back the configuration.

**Review by rnveach — CHANGES_REQUESTED** [review]

**Comment by romani** [comment]
https://app.travis-ci.com/github/checkstyle/eclipse-cs/jobs/588205391#L800
```
[ERROR] src/net/sf/eclipsecs/ui/CheckstyleUIPluginImages.java:[107] (extension) LineLengthExtended: Line is longer than 100 characters. Current line length is 114.
```

File: net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/CheckstyleUIPluginImages.java (hunk 1 above)
Comment (romani) [reply net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/CheckstyleUIPluginImages.java:107]: @Bananeweizen, please follow checkstyle

Comment (Bananeweizen) [reply net.sf.eclipsecs.ui/src/net/sf/eclipsecs/ui/CheckstyleUIPluginImages.java:107]: @rnveach @romani All the many commits of all my recent PRs _are_ free of checkstyle violations locally. Both in eclipse-cs as well as when running "mvn clean install" on the repository root. I've verified that on 2 different machines. You may want to verify on your local machines maybe. I believe the CI system still gives different results than a local build, even with the patch-filter changes.

**Review by romani — APPROVED** [review]
ok to merge.

thanks a lot for keeping CI green

**Pull request merged**